java/com/jfreechartextensions/axistransformations/ExtendedNumberAxis.java

```diff
@@ -372,38 +386,38 @@ protected void autoAdjustRange() {
                 if(this.getRangeType() == RangeType.POSITIVE) {
                     lower = Math.max(0.0D, lower);
                     upper = Math.max(0.0D, upper);
-                } else if(this.getRangeType() == RangeType.NEGATIVE) {
+                } else if (this.getRangeType() == RangeType.NEGATIVE) {
                     lower = Math.min(0.0D, lower);
                     upper = Math.min(0.0D, upper);
                 }
 
-                if(this.getAutoRangeIncludesZero()) {
+                if (this.getAutoRangeIncludesZero()) {
                     lower = Math.min(lower, 0.0D);
                     upper = Math.max(upper, 0.0D);
                 }
 
-                final double range = upper - lower;
+                final double range = Math.abs(upper - lower);
```

Comment:
If upper and lower are properly ordered, shouldn't this abs be unnecessary?

Reply:
Looks like the only way this happens is when the transform makes the lower > upper or vice versa. Is this correct? Should we support transforms where that's true?

Reply:
If upper and lower are properly ordered, then this is not necessary. However, this Math.abs() makes sure that the range is always positive as it is needed as the positive number (something like length which can't be negative)

Devon, yes. Currently, we have decided that we are not supporting those. We'll look into those as per customer's requests.

Should I make it as it was earlier? (Without Math.abs)

Reply:
We absolutely should support order inverting transforms.  There is no reason such should not work.